### ISSUE TITLE
Labels to cuda

### ISSUE THREAD

**Issue by Rusteam** [KaiyangZhou/deep-person-reid#136]
I followed your docs on creating [custom dataset](https://kaiyangzhou.github.io/deep-person-reid/user_guide.html#use-your-own-dataset) using this format  [(img_path, pid, camid), ...], registered and initialized a datamanager.

However on training it seems like it tries to move a python list of labels to cuda [here](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/engine/image/softmax.py) on line 104: ``pids = pids.cuda()``

Should not it be converted to torch tensor before cuda? It throws me an error

**Comment by KaiyangZhou** [comment]
In the [__getitem__](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/data/datasets/dataset.py#L226), `pid` is an int, so the batched `pids` should has been converted to `torch.LongTensor`.

what is the data type of your `pid`?

**Comment by Rusteam** [comment]
Oh gotcha. 

Mine were ‘0001’, ‘0002’, etc

**Comment by KaiyangZhou** [comment]
cool

you can have a look at [this pytorch code](https://github.com/pytorch/pytorch/blob/master/torch/utils/data/_utils/collate.py#L31), which details how each variable's dtype is changed during the batching process
